### DISCUSSION TITLE
Changing input file parameter from command line

### DISCUSSION THREAD

**Discussion by aaelmeli** [idaholab/moose#20085] [Q&A General] [answered]
Hi 
In an optimization problem, I am using Matlab script to run moose on wsl. I am using Matlab only to perform optimization while the actual simulation is done through moose. This is fine as long as the input file/s is not changing during the optimization process. Now, I need to loop over a parameter that requires changing kernel parameters as well as the boundary condition parameters within the input file.  For example, here is an example of what parameter do I need to modify in kernel and boundary conditions. Here, `rate` and `coef `are the parameters. As a note, let assume that the parameter that is changing is `freq`, then, `coef= some factor * freq` and `rate = freq^2`
```
[reaction_imagy]
        type = Reaction
        variable = uyi
        rate = -10000 this is the **(parameter)^2.**
    []

  [uxr_left]
        type = CoupledVarNeumannBC
        variable = uxr
        boundary = '3'
        v = uxi
        coef=-100 this is the parameter or a some scaling of the parameter
  []
```
So, how can I achieve this? Can I change an input parameter from the command line? Do I have other options? Also, How can I relate a parameter in the kernels section to a parameter in the boundary condition section?

Any suggestions?

**Comment by GiudGiud** [comment, marked as answer]
Hello 

You can change input parameters from the command line. 
for example here: ./app-opt -i input_file.i reaction_imagy/rate=-10002 uxr_left/coef=-101

You can do the second thing by defining it at the top of the input file:

value_to_use = 45
and then in the kernel and in the BC:

```
[reaction_imagy]
        type = Reaction
        variable = uyi
        rate = ${value_to_use}
    []

  [uxr_left]
        type = CoupledVarNeumannBC
        variable = uxr
        boundary = '3'
        v = uxi
        coef=${value_to_use}
  []

```

**Reply by aaelmeli** [reply]
> Hello
> 
> You can change input parameters from the command line. for example here: ./app-opt -i input_file.i reaction_imagy/rate=-10002 uxr_left/coef=-101
> 
> You can do the second thing by defining it at the top of the input file:
> 
> value_to_use = 45 and then in the kernel and in the BC:
> 
> ```
> [reaction_imagy]
>         type = Reaction
>         variable = uyi
>         rate = ${value_to_use}
>     []
> 
>   [uxr_left]
>         type = CoupledVarNeumannBC
>         variable = uxr
>         boundary = '3'
>         v = uxi
>         coef=${value_to_use}
>   []
> ```

Thank you, @GiudGiud . 

And I assume that I can multiply the `value_to_use `by some number to scale it or raise it to a power, Am I correct?
something like:
`
rate=(${value_to_use})^2`
and
`coef=0.76 * ${value_to_use}`

**Reply by GiudGiud** [reply]
You can but it takes a different syntax.

`${fparse value_to_use*value_to_use}`

**Reply by GiudGiud** [reply]
here's the doc page for all this:
https://mooseframework.inl.gov/application_usage/input_syntax.html
